Blahgger/Blahgger/Data/BlahggerData.cs

```diff
@@ -0,0 +1,137 @@
+﻿using Blahgger.Models;
+using System.Data.SqlClient;
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Web.Security;
+
+namespace Blahgger.Data
+{
+    /// <summary>
+    /// This class contains methods for binding data from the database to models.
+    /// </summary>
+    public class BlahggerData
+    {
+        private static BlahggerData singleton = null;
+
+        private BlahggerData() { }
+
+        public static BlahggerData GetInstance()
+        {
+            if (singleton == null)
+            {
+                singleton = new BlahggerData();
+            }
+            return singleton;
+        }
+        
+        public void AddUser(User user)
+        {
+            string sql = @"
+                INSERT INTO Users (Username, HashedPassword)
+                VALUES (@Username, @HashedPassword)
+            ";
+            // TODO Password hashing
+
+            DatabaseHelper.Insert(sql,
+                new SqlParameter("@Username", user.Username),
+                new SqlParameter("@HashedPassword", user.HashedPassword));
+
+        }
+
+        internal int GetUsersId(string username)
+        {
+            string sql = @"
+                SELECT Id
+                FROM Users
+                WHERE Username=@Username
+            ";
+
+            DataTable userTable = DatabaseHelper.Retrieve(sql,
+                new SqlParameter("@Username", username));
+            if (userTable.Rows.Count == 1)
+            {
+                return userTable.Rows[0].Field<int>("Id");
+            }
+            else
+            {
+                throw new Exception("SELECT query returned unexpected number of rows");
+            }
+        }
+
+        /// <summary>
+        /// This method checks whether or not a user's credentials match a row in the database.
+        /// </summary>
+        /// <param name="user">The input user object. This is assumed to have non-null Username and HashedPassword properties.</param>
+        /// <returns>True if a match is found; false otherwise.</returns>
+        internal bool AuthenticateUser(User user)
```

Comment:
I'd separate the retrieval of the user's hashed password from the authentication-related code. Either do the authentication portion in the controller or add another security-related class.